Konteh/Konteh.BackOfficeApi/Features/Questions/GetQuestionById.cs

```
@@ -3,65 +3,64 @@
-        }
+        public long Id { get; set; }
+    }
 
-        public class Response
-        {
-            public long Id { get; set; }
-            public string Text { get; set; } = string.Empty;
-            public QuestionCategory Category { get; set; }
-            public QuestionType Type { get; set; }
-            public List<AnswerDto> Answers { get; set; } = [];
-        }
+    public class Response
+    {
+        public long Id { get; set; }
+        public string Text { get; set; } = string.Empty;
+        public QuestionCategory Category { get; set; }
+        public QuestionType Type { get; set; }
+        public List<AnswerDto> Answers { get; set; } = [];
+    }
 
-        public class AnswerDto
+    public class AnswerDto
+    {
+        public long Id { get; set; }
+        public string Text { get; set; } = string.Empty;
+        public bool IsCorrect { get; set; }
+        public bool IsDeleted { get; set; }
+    }
+
+    public class RequestHandler : IRequestHandler<Query, Response>
+    {
+        private readonly IRepository<Question> _repository;
+
+        public RequestHandler(IRepository<Question> repository)
         {
-            public long Id { get; set; }
-            public string Text { get; set; } = string.Empty;
-            public bool IsCorrect { get; set; }
-            public bool IsDeleted { get; set; }
+            _repository = repository;
         }
 
-        public class RequestHandler : IRequestHandler<Query, Response>
+        public async Task<Response> Handle(Query request, CancellationToken cancellationToken)
         {
-            private readonly IRepository<Question> _repository;
+            var question = await _repository.GetById(request.Id);
 
-            public RequestHandler(IRepository<Question> repository)
+            if (question == null)
             {
-                _repository = repository;
+                throw new Exception("Question not found.");
```

should be NotFoundException. Also check for other places where we can throw that exception